src/session/session_api.c

```diff
@@ -273,8 +273,14 @@ __session_close(WT_SESSION *wt_session, const char *config)
     F_CLR(session, WT_SESSION_CACHE_CURSORS);
 
     /* Rollback any active transaction. */
-    if (F_ISSET(session->txn, WT_TXN_RUNNING))
+    if (F_ISSET(session->txn, WT_TXN_RUNNING)) {
+
+        /* Fix any eviction thread transaction. */
+        if (F_ISSET(session, WT_SESSION_INTERNAL) && !WT_SESSION_IS_CHECKPOINT(session))
+            WT_SESSION_TXN_SHARED(session)->pinned_id = WT_TXN_NONE;
```

Comment:
Same explanation in a comment of why we need to reset the pinned_id before rollback?

Reply:
Fix in code that start transaction means we don't need this anymore.